### ISSUE TITLE
在语义分割任务中,train.py 或 inference.py中, 我没有看到有关IoU计算的相关代码, 以及 inference.py代码存在一个小bug

### ISSUE THREAD

**Issue by FhilMs** [walking-shadow/Official_Remote_Sensing_Mamba#13]
inference代码中的bug如下:        
    for batch_img1, labels, name in tqdm(test_loader):
    ...
        ss_preds = torch.sigmoid(ss_preds[0]) #我将其改为
        ss_preds = torch.sigmoid(ss_preds) 
改成下面那行代码后 才可以正常推理, 原代码对ss_preds切片, 会导致缺失维度,使得预测与标签维度不匹配.
最后希望作者说明您是如何计算IoU的, 因为我在您论文中看到您将IoU看作了一个重要的评价指标.

**Comment by FhilMs** [comment]
> 好的，我修改一下inference.py中的代码。 IoU是在Precision和Recall的基础上进行计算的，计算的公式你可以自己搜索一下，很容易就可以搜到。

您好, 语义分割部分中, 我在查看由train.py生成的wandb图表, 对如下两个图表有些困惑
![image](https://github.com/walking-shadow/Official_Remote_Sensing_Mamba/assets/108119699/99e0d1b5-268b-4596-845c-eef54ae8b19f) 一个是epoch_val_f1score 一个是 val_f1score
这两者的区别在于什么呢?
根据您的代码, 
第二个图表数据
        log_wandb.log({
            f'{mode} loss': cd_loss,
            f'{mode} accuracy': batch_metrics['accuracy'],
            f'{mode} precision': batch_metrics['precision'],
            f'{mode} recall': batch_metrics['recall'],
            f'{mode} f1score': batch_metrics['f1score'],
            'learning rate': optimizer.param_groups[0]['lr'],
            f'{mode} loss_dice': diceloss,
            f'{mode} loss_bce': foclaloss,
            'step': total_step,
            'epoch': epoch
        })
是在单次epoch循环的enumerate 迭代过程中记录的
而 第一张图表对应的数据
    for k in epoch_metrics.keys():
        log_wandb.log({f'epoch_{mode}_{str(k)}': epoch_metrics[k],
                       'epoch': epoch})  # log epoch metric
是在单次epoch循环 结束enumerate迭代后记录的
